index.js

```diff
@@ -14,11 +15,19 @@ class Compat {
    *  @param {String} [options.rangeKey] The rangeKey for the model
    *  @param {Function} [options.createKey] The function to create the hashKey for certain models
    */
-  constructor({ model, hashKey = 'key', rangeKey, createKey }) {
+  constructor({ model, hashKey = 'key', rangeKey, createKey, ...fns }) {
     this.model = model;
     this.hashKey = hashKey;
     this.rangeKey = rangeKey;
     this._createKey = createKey;
+    // The idea of what we are doing here is to enable a way for functions to be
+    // added to this class while also being able to track their name so other wrappers
+    // like AwaitWrap can auto hoist them as well
+    this.hoistable = Object.keys(fns);
+    this.hoistable.forEach((key) => {
+      this[key] = fns[key];
+    });
```

Comment:
We don't have to worry about non-node environments here right? Can we just do `Object.assign(this, fns)` instead of this `forEach`

Reply:
that should be fine